compiler/src/dotty/tools/dotc/typer/Inferencing.scala

```diff
@@ -509,6 +509,8 @@ trait Inferencing { this: Typer =>
         //     required: String
         //     val y: List[List[String]] = List(List(1))
         val hasUnreportedErrors = state.reporter.hasUnreportedErrors
+        if hasUnreportedErrors then return tree
```

Comment:
@odersky Any reason we previously still instantiated some type variables when hasUnreportedErrors is true?

Reply:
I can't see a reason.